client/src/components/assessments/utils.js

```diff
@@ -0,0 +1,33 @@
+import PropTypes from "prop-types"
+
+const PERIOD_START_SHORT_FORMAT = "D"
+const PERIOD_START_MIDDLE_FORMAT = "D MMMM"
+const PERIOD_START_LONG_FORMAT = "D MMMM YYYY"
+const PERIOD_END_FORMAT = "D MMMM YYYY"
+
+export const periodToString = period => {
+  if (period.start.isSame(period.end, "day")) {
+    return period.end.format(PERIOD_END_FORMAT)
+  } else {
+    const periodStartFormat =
+      period.start.year() !== period.end.year()
+        ? PERIOD_START_LONG_FORMAT
+        : period.start.month() !== period.end.month()
+          ? PERIOD_START_MIDDLE_FORMAT
+          : PERIOD_START_SHORT_FORMAT
+    return `${period.start.format(periodStartFormat)} - ${period.end.format(
+      PERIOD_END_FORMAT
+    )}`
```

Comment:
`period.start` and `period.end` could be null/undefined; also, they appear to be of type `moment`, not simply `object`.

Reply:
This file was actually not being used, but the same code was also in periodUtils.js
Added react-moment-proptypes, and used it to do moment typechecking.
Updated the periods related proptypes and made the start and end properties required. Also marked other props in proptypes required.